### DISCUSSION TITLE
KB4011/KB6011: Support

### DISCUSSION THREAD

**Discussion by LeonisX** [Gissio/radpro#140] [Ideas]
I recently received the KB6011 detector by mail and was initially very skeptical about its capabilities. But as I study the documentation, I am more and more imbued with its capabilities and the potential that is embedded in it.

**Brief summary:**

Geiger-Muller counter + PIN diode. Registers Gamma (γ), hard Beta (β) and X-rays.

- Unique features: support for several types of Geiger counters, the ability to manually calibrate and adjust the voltage. There is a factory calibration function. Original technological solutions to improve the accuracy of measurements. Backlight settings are not saved, after switching on the screen is always orange (otherwise the light might be turned off).
- Pros: several ways to output data, a wide measurement range (up to 5000 μSv/h), an economical screen, small size, neat design, the ability to update via USB Type-C. Cost up to $ 50.
- Cons: not the most durable plastic case, no tightness, no strap mount, poor interface customization options, no function for saving and transferring results to a PC.

The developers publish the latest firmware versions and software for updating the detector.

I would like to know - is there potential for improvement in the firmware of this detector? So far I see that rather clumsy fonts are used and hieroglyphs slip through here and there.

Extended documentation in Chinese (you can translate the text in Google Translate).

https://h5.clewm.net/?url=qr61.cn%2Fo8FLvW%2Fqg7oWj8
https://h5.clewm.net/?url=http%3A%2F%2Fqr61.cn%2Fo8FLvW%2Fq5NoI3q

[2024 KB6011说明书.pdf](https://github.com/user-attachments/files/17019836/2024.KB6011.pdf)
[Manual KB4011-KB6011.pdf](https://github.com/user-attachments/files/17019835/Manual.KB4011-KB6011.pdf)

**Comment by Gissio** [comment]
Hi LeonisX:

I don't own a KB4011/KB6011, so I can't support it. But I can give you some hints as to what would need to be done (general hints, not support):

First, find out what microprocessor is in your device: most likely it is an STM32 or clone. Follow the traces from the microprocessor to the board's devices (keyboard, display, battery, etc) and see what port connects to what device. A multimeter might help. You should also consider using a disassembly tool to disassemble the firmware, and see how the code accesses the microprocessor's internal devices (input/output ports, timers, ADCs, etc).

Finally, follow the steps in https://github.com/Gissio/radpro/blob/main/docs/developers.md and customize the code for your device.

This should make a nice project...

**Comment by LeonisX** [comment]
Yes, perhaps it will be an interesting study, if I find time for it.

The detector manufacturer shows in the screenshots that it is: GD32E230C8T6.

https://www.gigadevice.com/product/mcu/mcus-product-selector/gd32e230c8t6

I have not yet launched GigaDevice_MCU_ISP_Programmer to verify this. The latest firmware can be downloaded in HEX format, I found how to convert it to DFU, but it is probably better to work with the BIN format, with clean data without any headers. I do not know yet how to remove the header without manipulations in the HEX editor.

I worked with assembler, although I do not have much experience. It is worse with microcontrollers, I started studying Arduino, but I have not assembled anything in practice yet.

In general, I thought about disassembling the firmware and then decompiling it into C, but this is only in theory. I do not want to completely redo the interface. It would be nice to improve the fonts, add support for other languages, improve energy saving by turning off the screen completely, and so on. Almost 14 KB can be added. There are other ideas, but they depend on how complex the calculation logic is. I think the developers have overworked themselves.

Thanks for the advice :)

**Reply by Gissio** [reply]
It sounds like you have what it takes to tackle this problem!

**Comment by Weesky** [comment]
We have launched our official firmware page on GitHub. If you have any questions related to our products, you can provide feedback here. If you have any suggestions for improvement, we recommend leaving messages in the Issues section. Within the next few months, we will be releasing the latest 4.5 firmware for the KB6011 version 24.

We are currently working on improving the multilingual interface. Stay tuned.

https://github.com/Weesky/KB6011/releases/tag/v4.3.0
https://github.com/Weesky/KB6011

**Reply by Montecri** [reply]
Hi! Great initiative!

Is there an official Aliexpress link?

Thanks!

**Reply by Weesky** [reply]
> Hi! Great initiative!
> 
> Is there an official Aliexpress link?
> 
> Thanks!

https://www.aliexpress.com/item/2251832871987194.html
